lib/ash/subject.ex

```diff
@@ -0,0 +1,539 @@
+  end
+
+  @doc """
+  Sets the actor for the subject.
+
+  The actor represents the user or entity performing the action.
+  Can be set directly or via keyword list with `:actor` key.
+
+  ## Parameters
+
+    * `subject` - The subject to set actor on
+    * `actor_or_opts` - Either the actor directly, or keyword list with `:actor` key
+  """
+  @spec set_actor(t(), Keyword.t() | term()) :: t()
+  def set_actor(subject, opts) when is_list(opts) do
+    if Keyword.has_key?(opts, :actor) do
+      set_actor(subject, opts[:actor])
+    else
+      subject
+    end
+  end
+
+  def set_actor(subject, actor) do
+    put_context(subject, :private, %{actor: actor})
+  end
+
+  @doc """
+  Sets the tenant for the subject.
+
+  Used for multi-tenant applications to specify which tenant the operation
+  should be performed for.
+
+  ## Parameters
+
+    * `subject` - The subject to set tenant on
+    * `tenant_or_opts` - Either the tenant directly, or keyword list with `:tenant` key
+  """
+  @spec set_tenant(t(), Keyword.t() | term()) :: t()
+  def set_tenant(subject, opts) when is_list(opts) do
+    if Keyword.has_key?(opts, :tenant) do
+      set_tenant(subject, opts[:tenant])
+    else
+      subject
+    end
+  end
+
+  def set_tenant(subject, tenant) do
+    %{subject | tenant: tenant, to_tenant: Ash.ToTenant.to_tenant(tenant, subject.resource)}
+  end
+
+  @doc """
+  Sets the authorization flag for the subject.
+
+  Controls whether authorization checks should be performed.
+
+  ## Parameters
+
+    * `subject` - The subject to set authorization on
+    * `authorize_or_opts` - Either boolean directly, or keyword list with `:authorize?` key
+  """
+  @spec set_authorize?(t(), Keyword.t() | term()) :: t()
+  def set_authorize?(subject, opts) when is_list(opts) do
+    if Keyword.has_key?(opts, :authorize?) do
+      set_authorize?(subject, opts[:authorize?])
+    else
+      subject
+    end
+  end
+
+  def set_authorize?(subject, authorize?) do
+    put_context(subject, :private, %{authorize?: authorize?})
+  end
+
+  @doc """
+  Sets the tracer for the subject.
+
+  Used for tracing and monitoring operations.
+
+  ## Parameters
+
+    * `subject` - The subject to set tracer on
+    * `tracer_or_opts` - Either the tracer directly, or keyword list with `:tracer` key
+  """
+  @spec set_tracer(t(), Keyword.t() | term()) :: t()
+  def set_tracer(subject, opts) when is_list(opts) do
+    if Keyword.has_key?(opts, :tracer) do
+      set_tracer(subject, opts[:tracer])
+    else
+      subject
+    end
+  end
+
+  def set_tracer(subject, tracer) do
+    put_context(subject, :private, %{tracer: tracer})
+  end
+
+  @doc """
+  Sets a result value in the subject's private context.
+
+  Used to store action results for later retrieval.
+
+  ## Parameters
+
+    * `subject` - The subject to set result on
+    * `result` - The result data to store
+  """
+  @spec set_result(t(), term()) :: t()
+  def set_result(subject, result) do
```

Comment:
We don't support this on generic actions yet, so we either need to support it there or remove this here.